Prototype code for /mail to get sent via Discord

RhysB opened the pull request.

RhysB committed:
Prototype code for /mail to get sent via Discord

Grassboii commented on Essentials/src/main/resources/plugin.yml:
```diff
@@ -183,7 +183,7 @@ commands:
   mail:
     description: Manages inter-player, intra-server mail.
-    usage: /<command> [read|clear|send [to] [message]]
+    usage: /<command> [read|clear|send|discord] [to] [message]]
```
extra ']' at the end

zavdav commented on Essentials/src/main/java/com/earth2me/essentials/UserData.java:
```diff
@@ -617,4 +620,48 @@ public void setNPC(boolean set) {
         config.save();
     }
+
+    public void setUUUID(UUID uuid) {
```
setUUUID()

Garsooon commented:
Works on local server. It might be good to return a help case if discord is applied in the regular command ie.
/mail discord Garsooon test or /mail send discord Garsooon test> "Proper usage is /mail discord to toggle discord mail!" or something of the like
otherwise code looks fine

Garsooon committed:
Remove ]

Garsooon reviewed (dismissed):
see previous comment for what I think might be beneficial

Garsooon committed:
Resolve stale pr conflicts

Garsooon committed:
Merge branch 'master' into Discord-Mail

Garsooon committed:
Versioning

Garsooon committed:
Merge remote-tracking branch 'origin/Discord-Mail' into Discord-Mail

# Conflicts:
#	Essentials/pom.xml
#	EssentialsChat/pom.xml
#	EssentialsGeoIP/pom.xml
#	EssentialsProtect/pom.xml
#	EssentialsSpawn/pom.xml
#	pom.xml

Garsooon committed:
Fix merge error

RhysB committed:
Enhances Discord mail integration

Improves the Discord mail forwarding feature by:

- Ensuring mail forwarding only occurs if both Discord plugins are enabled.
- Preventing forwarding to online players.
- Preventing forwarding to players who have disabled the feature.
- Adds logging for successful and failed Discord mail attempts.
- Fixes a typo in the UserData class that was preventing UUID's being set correctly.
